app/javascript/admin/controllers/config_controller.js

```diff
@@ -45,7 +45,7 @@ export default class ConfigController extends Controller {
     }
   }
 
-  closeAdminConfigModal() {
+  closeAdminModal() {
```

Comment:
```suggestion
  yourFunctionHere() {
    // I think you'll want to use fetch() here to make a request to the ConfigsController#create action, which is what happens when you click on the "Update Site Configuration" button
    // Something like this...
    fetch(`/admin/config`, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'X-CSRF-Token': document.querySelector("meta[name='csrf-token']")
          .content,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        // I think the request will fail without the confirmation text 😔 
        confirmation: '...'
      }),
      credentials: 'same-origin',
      })
  },

  closeAdminModal() {
```

This will be a little tricky because of all the authorization stuff we have in place when someone tries to update the SiteConfig. I'm not sure that the pseudocode I've thrown in here will work as-is, but that is effectively what you'll need to do if you want to submit a request to the backend from this JS button.

Take a look at [this function](https://github.com/forem/forem/blob/master/app/javascript/admin/controllers/reaction_controller.js#L10-L22), which is kind of similar/along the same lines.

Reply:
Instead of fetch, use our `request` utility function. 

```javascript
fetch('/admin/config', {
  method: 'POST',
  headers: {
    Accept: 'application/json',
    'X-CSRF-Token': document.querySelector("meta[name='csrf-token']").content,
    'Content-Type': 'application/json',
  },
  body: JSON.stringify({
    // I think the request will fail without the confirmation text 😔
    confirmation: '...',
  }),
  credentials: 'same-origin',
});
```

would become this

```javascript
request('/admin/config', {
  method: 'POST',  
  body: { // Note: you do not need to do JSON.stringify. The request function handles that.
    // I think the request will fail without the confirmation text 😔
    confirmation: '...',
  }
})
```

Reply:
Oops, I think the comment I commented out is no longer relevant. 😅